mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -862,8 +881,11 @@ public void forEachTuple(ModelConsumer consumer, TupleTypeContext tupleTypeConte
 		}
 
 		String collectionName = getCollectionName( backendQuery, queryDescriptor, entityKeyMetadata );
+		//WriteConcern writeConcern = getWriteConcern( tupleContext );
+		//ReadPreference readPreference = getReadPreference( tupleContext );
 		MongoCollection<Document> collection = provider.getDatabase().getCollection( collectionName );
-
```

Comment:
How come you area not applying the withOption method in this case? You are not applying anymore the write or read options, I'm surprised that no test fails in because of this

Reply:
Applying options when executing backend query is OGM-1448 bug. In tests only getObject, getObjects and getAssociation are called

Reply:
Mmmh... OK